### ISSUE TITLE
More realistic precipitation and temperature

### ISSUE THREAD

**Issue by Kodiologist** [CleverRaven/Cataclysm-DDA#34139]
**Is your feature request related to a problem? Please describe.**

I've checked some real weather data from New England and found that temperature is much more variable day-to-day (about 5×) than it is in Cataclysm, and that precipitation is much less common (about 1/5th) and less intense (about 1/10th).

**Describe the solution you'd like**  

I'd like to make temperature more variable and lower the rate of preciptation events. I intend to make these changes myself. This issue is just to check whether those changes would be welcome.

**Describe alternatives you've considered**

Is the temperature supposed to be less variable, or precipitation supposed to be more common, than in real life? I'd presume not, now that we're saying the game takes place in the present day instead of the 2040s (so there's no climate change to extrapolate to), and I don't think any of the backstory events (e.g., the goo invasion) should be changing the weather.

Cataclysm also simulates pressure and (relative) humidity, and my informal checks suggest that these are also unrealistic. (In particular, Cata simulates a temperature-like seasonal trend for pressure, whereas real pressure doesn't change much on a seasonal basis.) However, my chief concern is temperature and precipitation, because those are what have direct gameplay consequences. So, getting pressure and humidity to be realistic isn't that important, although I might change them as part of changing precipitation (given that Cata determines the precipitation using pressure and humidity).

**Additional context**

I've written [some data-extraction and analysis code](https://gist.github.com/Kodiologist/e2f1e6685e8fd865650f97bb6a67ad07). Here's what I find for three New England sites in 2014, compared to one year of Cata weather (actually winter of one year plus the other three seasons of the next year, to line up with the Gregorian calendar).

    Proportion of days with nonzero precipitation
    CONCORD, NH - 0.499
    BOSTON, MA - 0.482
    CARIBOU, ME - 0.658
    Cata - 0.973

    Proportion of hours with nonzero precipitation
    CONCORD, NH  - 0.151
    BOSTON, MA - 0.156
    CARIBOU, ME - 0.212
    Cata - 0.587

    Total rain per year in inches
    CONCORD, NH - 45.983
    BOSTON, MA - 45.311
    CARIBOU, ME - 45.179
    Cata - 570.446

(The total-rain figure for Cata is actually an underestimate because it doesn't count the rain equivalent of snow, whereas the real data does.)

Here are daily highs (top) and lows (bottom) in Concord, NH (blue) and Cataclysm (orange). Temperatures are in degrees Fahrenheit. Observe how Cataclysm roughly matches the magnitude and seasonal trend of the real temperatures, but is much more consistent from day to day.

![highs](https://i.imgur.com/ayhPlBq.png)

![lows](https://i.imgur.com/zrJJi4J.png)

**Comment by KorGgenT** [comment]
i've seen the graphs used when making the weather patterns. it's for Massachusetts.

**Comment by KorGgenT** [comment]
@davidpwbrown if you have any additional comment

**Comment by kevingranade** [comment]
Yes we'd like to match real-world weather trends, this would be very
welcone, especially if it came with some validation code.

**Comment by Amoebka** [comment]
Increasing daily temperature variation runs the risk of greatly increasing tedium. If the player has to change clothing every evening/morning, that's a lot of extra clicks for very little immersion benefit.

**Comment by Kodiologist** [comment]
I've been working on this and I'm pretty happy with how temperature, pressure, and humidity now look. Next, precipitation.

**Comment by Kodiologist** [comment]
#34515 is complete (in theory) and should fix this.
